multiversx_sdk_core/transaction_factories/relayed_transactions_factory.py

```diff
@@ -0,0 +1,108 @@
+import base64
+import json
+from typing import Any, Dict, List, Protocol
+
+from multiversx_sdk_core import Address
+from multiversx_sdk_core.errors import InvalidInnerTransactionError
+from multiversx_sdk_core.interfaces import IAddress, ITransaction
+from multiversx_sdk_core.serializer import args_to_string
+from multiversx_sdk_core.transaction import Transaction
+
+
+class IConfig(Protocol):
+    chain_id: str
+    min_gas_limit: int
+    gas_limit_per_byte: int
+
+
+class RelayedTransactionsFactory:
+    def __init__(self, config: IConfig) -> None:
+        self._config = config
+
+    def create_relayed_v1_transaction(self,
+                                      inner_transaction: ITransaction,
+                                      relayer_address: IAddress) -> Transaction:
+        if not inner_transaction.gas_limit:
+            raise InvalidInnerTransactionError("The gas limit is not set for the inner transaction")
+
+        if not inner_transaction.signature:
+            raise InvalidInnerTransactionError("The inner transaction is not signed")
+
+        serialized_transaction = self._prepare_inner_transaction_for_relayed_v1(inner_transaction)
+        data = f"relayedTx@{serialized_transaction.encode().hex()}"
+
+        gas_limit = self._config.min_gas_limit + self._config.gas_limit_per_byte * len(data) + inner_transaction.gas_limit
+
+        return Transaction(
+            chain_id=self._config.chain_id,
+            sender=relayer_address.to_bech32(),
+            receiver=inner_transaction.sender,
+            gas_limit=gas_limit,
+            data=data.encode()
+        )
+
+    def create_relayed_v2_transaction(self,
+                                      inner_transaction: ITransaction,
+                                      inner_transaction_gas_limit: int,
+                                      relayer_address: IAddress) -> Transaction:
+        if inner_transaction.gas_limit:
+            raise InvalidInnerTransactionError("The gas limit should not be set for the inner transaction")
+
+        if not inner_transaction.signature:
+            raise InvalidInnerTransactionError("The inner transaction is not signed")
+
+        arguments: List[Any] = [
+            Address.new_from_bech32(inner_transaction.receiver),
+            inner_transaction.nonce,
+            inner_transaction.data,
+            inner_transaction.signature
+        ]
+
+        data = f"relayedTxV2@{args_to_string(arguments)}"
+        gas_limit = inner_transaction_gas_limit + self._config.min_gas_limit + self._config.gas_limit_per_byte * len(data)
+
+        return Transaction(
+            sender=relayer_address.to_bech32(),
+            receiver=inner_transaction.sender,
+            amount=0,
+            gas_limit=gas_limit,
+            chain_id=self._config.chain_id,
+            data=data.encode(),
+            version=inner_transaction.version,
```

Comment:
:+1:

Reply:
Correct, with respect to: https://github.com/multiversx/mx-sdk-js-core/issues/367.